ue4docker/infrastructure/BuildConfiguration.py

```diff
@@ -169,7 +172,7 @@ def addArguments(parser):
         )
         parser.add_argument(
             "--visual-studio",
-            default=VisualStudio.VS2017,
+            default=VisualStudio.VS2019,
```

Comment:
Please, no. We're supporting engines back to 4.20 and those do not support VS2019. And Epics switched to VS2019 only in UE5, so this all is a very modern stuff.

We possibly could add some more sophisticated default selection logic that would choose VS2017 by default for UE4 and VS2019 for UE5.

Using VS2022 by default is a no-no until Epics do that themselves. We're open to provide configurable setups, but default behavior should be as close to what Epics do as possible, to avoid introducing additional issues.

Reply:
VS2019 is default since 4.25 if their documentation doesn't lie.

Automatic default logic to mimic Epic's recommended Visual Studio would be nice, but not strictly necessary. A documentation note would be helpful though, so you don't wait hours for the build only to realize you should have used a newer VS version, because you assume it would use the "correct" one based on the Unreal version.

Switched back to VS2017 as default.

Reply:
> VS2019 is default since 4.25

There are two contradicting things: what Epics say and what Epics do. It is true that [this page](https://docs.unrealengine.com/4.26/en-US/ProductionPipelines/DevelopmentSetup/VisualStudioSetup/) claims VS2019 to be default since 4.25. Buuut! If you closely look at release notes of 4.25/4.26/4.27, you'll find out that launcher-distributed engine is built with VS2017.

Reply:
True. I also mixed up engine source build vs just using VS to package projects.

Reply:
There is also [this](https://github.com/EpicGames/UnrealEngine/blob/4.27/Engine/Build/InstalledEngineBuild.xml#L79). For installed builds (and ue4-docker produces installed builds), VS2017 is default in 4.x releases.

Reply:
Yes, but VS2022 should still work, it's backwards compatible enough that everything else works with VS2019 flags.